conan/internal/api/export.py

```diff
@@ -97,20 +97,21 @@ def _calc_revision(scoped_output, path, manifest, revision_mode, conanfile):
     if revision_mode == "hash":
         revision = manifest.summary_hash
     else:
+        repository = (revision_mode == "scm")
         # Exception to the rule that tools should only be used in recipes, this Git helper is ok
         excluded = getattr(conanfile, "revision_mode_excluded", None)
         git = Git(conanfile, folder=path, excluded=excluded)
         try:
-            revision = git.get_commit(repository=(revision_mode == "scm"))
+            revision = git.get_commit(repository=repository)
         except Exception as exc:
             error_msg = "Cannot detect revision using '{}' mode from repository at " \
                         "'{}'".format(revision_mode, path)
             raise ConanException("{}: {}".format(error_msg, exc))
 
-        if git.is_dirty():
-            raise ConanException("Can't have a dirty repository using revision_mode='scm' and doing"
+        if git.is_dirty(repository=repository):
```

Comment:
I have some concerns that this might be breaking for users that have the repository dirty with some temp files outside of the conanfile subfolder, and suddenly this stops working. Having those temporary files in the root repo wouldn't be a big problem in most cases, isn't it?

Is this check really that important to protect at the repository level?